### PR TITLE
[3.4] Revert 14548

### PR DESCRIPTION
**Pull request by mitake**
Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

https://github.com/etcd-io/etcd/pull/14322 was backported to release-3.4 in https://github.com/etcd-io/etcd/pull/14548, but it should be reverted too.

cc @ahrtr @serathius

### CODE REVIEW

**Comment by ahrtr** [comment]
@mitake  could you please also update the 3.4 changelog ?

**Review by ahrtr — APPROVED** [review]
LGTM

Thank you @mitake

**Comment by mitake** [comment]
Thanks @ahrtr @lavacat . I'll open a PR for updating the changelog tomorrow.

File: etcdserver/api/v3rpc/watch.go [hunk 1]
```diff
@@ -277,29 +277,13 @@ func (sws *serverWatchStream) recvLoop() error {
...
-				case auth.ErrUserEmpty:
-					cancelReason = rpctypes.ErrGRPCUserEmpty.Error()
-				default:
-					if err != auth.ErrPermissionDenied {
-						sws.lg.Error("unexpected error code", zap.Error(err))
-					}
-					cancelReason = rpctypes.ErrGRPCPermissionDenied.Error()
-				}
-
+			if !sws.isWatchPermitted(creq) {
```
Comment (ahrtr) [inline etcdserver/api/v3rpc/watch.go:280]: I think we can keep the server side change?

Comment (mitake) [reply etcdserver/api/v3rpc/watch.go:280]: ah sorry, let me open a follow up PR

Comment (mitake) [reply etcdserver/api/v3rpc/watch.go:280]: On the second thought, I guess just reverting the entire change might be safer? What do you think?

Comment (ahrtr) [reply etcdserver/api/v3rpc/watch.go:280]: Actually we kept the server side change when reverting the PR on 3.5.  Keeping the server side change can provide more useful error message (`cancelReason`).

But it isn't a big deal, it's up to you. thx

Comment (mitake) [reply etcdserver/api/v3rpc/watch.go:280]: got it, opened https://github.com/etcd-io/etcd/pull/15584

**Pull request merged**